File: 2.Add Two Numbers/AddTwoNumbers.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* l1l=l1;
        ListNode* l2l=l2;
        ListNode* tmp=NULL;
        bool get=false;
        int count=0;
        for(;l1l->next!=NULL;l1l=l1l->next){
            count++;
        }
        for(;l2l->next!=NULL;l2l=l2l->next){
            count--;
        }
        if(count<0){
            tmp=l1;
            l1=l2;
            l2=tmp;
            tmp=l1l;
            l1l=l2l;
            l2l=tmp;
        }//总之是要l1更长
        l1l->next=l2;
        ListNode* ll1=l1;
        ListNode* ll2=l2;
        while(ll1!=l2 && ll2!=NULL){
            ll1->val=ll2->val+ll1->val;
            printf("%d\n",ll1->val);
            if(get==true){
                ll1->val=ll1->val+1;
                get=false;
            }
            if(ll1->val>=10)
            {
                ll1->val=ll1->val%10;
                get=true;
            }
            ll1=ll1->next;
            ll2=ll2->next;
        }
        if(ll1==l2 && ll2==NULL)
        {
            if(get==true){
                ll1->val=1;
                ll1->next=NULL;
            }
            else{
                l1l->next=NULL;
            }
            return l1;
        }
        else if(ll1!=l2 && ll2==NULL){
            while(get==true && ll1!=l2){
                ll1->val++;
                get=false;
                if(ll1->val >= 10)
                {
                    get=true;
                    printf("%d\n",ll1->val);
                    ll1->val=ll1->val%10;
                    ll1=ll1->next;
                }
            }
            if(get==true && ll1==l2){
                ll1->val=1;
                ll1->next=NULL;
                return l1;
            }
            
            l1l->next=NULL;
        }
        
        return l1;
    }
};

File: 2.Add Two Numbers/AddTwoNumbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        carry=0
        head1=l1
        head2=l2
        
        while head1.next and head2.next:
            r=head1.val+head2.val+carry
            head1.val,carry=r%10,r/10
            head1=head1.next
            head2=head2.next
        val=head1.val+head2.val+carry
        head1.val,carry= val%10,val/10
        if carry and not head1.next and not head2.next:
            head1.next=l2
            l2.next=None
            l2.val=1
            return l1
        if head1.next:
            head1=head1.next
            while head1.next and carry:
                val=head1.val+carry
                head1.val,carry=val%10,val/10
                head1=head1.next
            if carry:
                val=head1.val+carry
                head1.val,carry=val%10,val/10
                if carry:
                    head1.next=l2
                    l2.next=None
                    l2.val=1
            return l1
        if head2.next:
            head2=head2.next
            head1.next=head2
            while carry and head2.next:
                val=head2.val+carry
                head2.val,carry=val%10,val/10
                head2=head2.next
            head1=head2
            if carry:
                val=head2.val+carry
                head2.val,carry=val%10,val/10
                if carry:
                    head2.next=l2
                    l2.next=None
                    l2.val=1
            return l1
        return l1
            